pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -260,6 +313,12 @@ func (nc *NodeController) SyncCloudNodes() error {
 			glog.Errorf("Delete node %s error: %v", nodeID, err)
 		}
 		nc.deletePods(nodeID)
+		nodesChanged = true
+	}
+
+	// Make external services aware of nodes currently present in the cluster.
+	if nodesChanged {
```

Comment:
ah, nm, I see that this is about adding things to the target pool, makes sense now.